### ISSUE TITLE
to name a cell should not re-fetch data?

### ISSUE THREAD

**Issue by tx0c** [observablehq/feedback#336]
usually when exploring a dataset, we're not starting all cells with names, but start many cells unnamed, like many `db.sql...` to get a piece of data; later on, some cells are found useful and need a name for plot graphs or something, we have to change it to `data = db.sql...` then it triggers a re-run the sql and re-fetching,

here is to request the UI has a way to name a cell, without re-run the cell, I suggest it can be in the menu like `name the cell?` and then it can pop a input box for user to type a name, then change the cell content to `name = ...original-content...`

for many reasons like:
1. the naming time-point might be some hours or days later, the original database has changed, re-run is returning a different result;
2. or just save the time of re-run, because some of my SQL is already requiring minutes to re-run

![image](https://user-images.githubusercontent.com/49659410/147891390-1183c21d-597a-4bdf-9993-06b70275aa52.png)

**Comment by CobusT** [comment]
Interesting suggestion! Curious about your 1st case though... do you typically keep your browser open on the notebook for days without refreshing?

**Comment by tx0c** [comment]
yes, and since the same notebook is growing bigger and bigger, might contain 100+ cells, then I found if close and re-open it, the initial loading would take many many minutes (or hours), or some would fail (might be server overloaded or other reasons);  can I request the initial opening of a notebook not to request all data at once? or have a RateLimitQueue behavior, like 5 concurrent queries at most, and/or 10 queries/second?

the alternative becomes the way I am currently using, because laptop nowadays support graceful suspend/resume, sometimes I keep the same browser same tabs not to close/re-open for months

**Comment by CobusT** [comment]
Sounds like a massive notebook. I would recommend controlling the loading of the various parts with functions that only query when you set flags. You can use a button to do this (or any other type of Input that works for your case. See this notebook for an example:
https://observablehq.com/d/8aaaa1f253017d47

**Comment by tx0c** [comment]
that button is another alternative; but if doing all 100+ cells with its own extra button and button wrapper, it's a lot of unnecessary dup code

here's requesting such feature should be builtin

**Comment by CobusT** [comment]
I think it is worth exploring not rerunning a cell when the cell name changes from unnamed to named.

**Comment by tx0c** [comment]
> changes from unnamed to named

and should also include renaming? generally:

from:

    expression

to:

    name1 = expression

or to

    name2 = expression

should not trigger re-running?  you have a parser can know the right side `expression` is not changing, then not to re-run?

especially with BigQuery / Snowflake, some queries can be very big and expensive

**Comment by tx0c** [comment]
> Sounds like a massive notebook

update for this:   these notebooks are 30 ~ 60 cells mostly, not massive
